modules/internal/ChapelArray.chpl

```diff
@@ -1473,6 +1485,98 @@ module ChapelArray {
     }
   }  // record _domain
 
+  proc __getActualInsertPts(d, inds, 
```

Comment:
I agree. This issue seems to be a slippery slope. As far as I can see ChapelArray.chpl has some functions that should be moved to other files/classes. eg. BaseSparseDom in ChapelDistribution.chpl is the most suitable class for this function to be IMO. But for consistency I'd like to keep it in ChapelArray(for now at least). I can come back to this at a larger refactoring context.